packages/web-components/fast-foundation/src/foundation-element/foundation-element.ts

```diff
@@ -216,43 +221,56 @@ export class FoundationElementRegistry<
         const prefix = definition.prefix || context.elementPrefix;
         const name = `${prefix}-${definition.baseName}`;
 
-        context.tryDefineElement(name, this.type, x => {
-            const presentation = new DefaultComponentPresentation(
-                resolveOption(definition.template, x, definition),
-                resolveOption(definition.styles, x, definition)
-            );
+        context.tryDefineElement({
```

Comment:
Should we Omit the `baseClass` from the `OverrideFoundationElementDefinition` as well?

Reply:
I'm not sure I follow the question 🤔

Reply:
Nevermind, just got the sense of the question (I need cafeine 😅 ). I think so yeah!

Reply:
Actually we can't `omit` it. `overrideDefinition` has the `OverrideFoundationElementDefinition` type
![image](https://user-images.githubusercontent.com/44816587/130130523-2780bffc-1469-4660-978b-9db2fdfbb0ba.png)

Reply:
I'm a bit confused because the code below does exactly this:

```ts
baseClass: overrideDefinition.baseClass
```

But it seems like that should never come from the override definition but only from the original definition.

Reply:
I think I see what you mean. The line in your previous comment is required as `baseClass` is passed along _twice_ in order to get from the `compose()` call all the way to the actual condition added in the other file changed in this PR.
I hope I was clear! :) However if I missed something definitely let me know!